addons/product_expiry/tests/test_stock_lot.py

```diff
@@ -615,3 +615,48 @@ def test_no_expiration_date(self):
         picking_out.action_confirm()
         picking_out.action_assign()
         self.assertEqual(picking_out.move_ids.state, 'assigned', 'Wrong state of move line.')
+
+    def test_no_lot(self):
+        """
+        Try to confirm a stock move with a quant that doesn't have lot
+        """
+        # Set the removal strategy to 'least_packages'
+        fefo_strategy = self.env['product.removal'].search(
+            [('method', '=', 'fefo')])
+        self.apple_product.categ_id.removal_strategy_id = fefo_strategy.id
+
+        lot_form = Form(self.LotObj)
+        lot_form.name = 'LOT001'
+        lot_form.product_id = self.apple_product
+        lot_form.company_id = self.env.company
+        apple_lot = lot_form.save()
+
+        self.StockQuantObj.with_context(inventory_mode=True).create([{
+            'product_id': self.apple_product.id,
+            'location_id': self.stock_location,
+            'quantity': 100,
+        },{
+            'product_id': self.apple_product.id,
+            'location_id': self.stock_location,
+            'quantity': 100,
+            'lot_id': apple_lot.id,
+        }])
+
+        picking_out = self.PickingObj.create({
+            'picking_type_id': self.picking_type_out,
+            'location_id': self.stock_location,
+            'location_dest_id': self.customer_location,
+            'state': 'draft',
+        })
+
+        self.MoveObj.create({
+            'name': self.apple_product.name,
+            'product_id': self.apple_product.id,
+            'product_uom_qty': 10,
+            'product_uom': self.apple_product.uom_id.id,
+            'picking_id': picking_out.id,
+            'location_id': self.stock_location,
+            'location_dest_id': self.customer_location
+        })
```

Comment:
```suggestion
        with Form(self.PickingObj) as picking_form:
            picking_form.picking_type_id = self.env.ref('stock.picking_type_out')
            with picking_form.move_ids_without_package.new() as move:
                move.product_id = self.apple_product
                move.product_uom_qty = 10
            picking_out = picking_form.save()
```
More concise this way, also avoids writing a lot of useless information. :wink: